junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -1993,6 +1993,24 @@ public static Set<Class<?>> getAllAssignmentCompatibleClasses(Class<?> clazz) {
 		return result;
 	}
 
+	/**
+	 * {@return whether the supplied {@code object} is an instance of a record class}
+	 * @since 1.12
+	 */
+	@API(status = INTERNAL, since = "1.12")
+	public static boolean isRecordObject(Object object) {
+		return object != null && isRecordClass(object.getClass());
+	}
+
+	/**
+	 * {@return whether the supplied {@code clazz} is a record class}
+	 * @since 1.12
+	 */
+	@API(status = INTERNAL, since = "1.12")
+	public static boolean isRecordClass(Class<?> clazz) {
+		return "java.lang.Record".equals(clazz.getSuperclass().getName());
```

Comment:
`clazz.getSuperclass()` can return `null`.

So, you'll need to defensively check for that to avoid a `NullPointerException`.

Reply:
Done in 58cb54143017afb945b4b6e67f5f069f34a1d735